python/delegates/view_item_delegate.py

```diff
@@ -3554,15 +3551,10 @@ def set_icon(self, icon):
 
         if not icon:
             self.icon = None
-
-        elif isinstance(icon, QtGui.QIcon):
```

Comment:
Why not replacing to `if hasattr(icon, "pixmap"):` like you did before instead of a "else" block

Reply:
I didn't use `hasattr(icon, "pixmap")` here since we don't actually use the `QIcon.pixmap` method.

This way puts the onus on the caller to pass the correct object for the icon, instead of checking if it is a valid icon and if not, setting the icon to None. Arguably this way might be better since if the icon is invalid, you'll get an error message down the line instead of silently just setting the icon to None.